### PR TITLE
Improve column number tracking

### PR DESCRIPTION
**Pull request by chqrlie**
- simplify column number tracking using a pointer to the beginning of line
  instead of `eol` + `mark`.
- add `js_parse_error_pos` to report syntax errors with correct column number
  for token parsing errors. This makes the syntax error reports much more precise.
- add `JSSourcePos` type to keep track of token source position
- add `emit_op_pos` to set the precise source position in code generation
  this is work in progress: it should be the only way to generate source positions
  and we should get rid of `s->last_line_num` and `s->last_col_num`.
- runtime errors on calls report the column number of calling function or method name.
- runtime errors on `new` expressions report the column number of the `neẁ` keyword.
- update **tests/test_builtin,js** with more informative messages
- improve `assert()` and **tests/test_language.js** tests
- update **v8.txt** for updated column numbers in remaining errors

### CODE REVIEW

**Comment by saghul** [comment]
I'll defer this review to @bnoordhuis who worked on adding column numbers :-P

File: v8.txt [hunk 1]
```diff
@@ -399,6 +399,26 @@ found:
 === json-errors.js
 Failure:
 expected:
+"Unexpected end of JSON input"
```
Comment (saghul) [inline v8.txt:402]: Are these new?

**Review by saghul — APPROVED** [review]
This needs a rebase, but it should be good to go otherwise. Sorry for the delay @chqrlie !

**Comment by TooTallNate** [comment]
Is this expected to improve the issue in https://github.com/quickjs-ng/quickjs/issues/236?

**Comment by chqrlie** [comment]
> Is this expected to improve the issue in #236?

Yes it should. Let me try this test after I fix the conflicts.

**Comment by ammarahm-ed** [comment]
Hi, I am working on a project that is blocked due to this, can we possibly get this merged anytime soon.

**Comment by saghul** [comment]
@chqrlie are you planning on picking this back up? It likely needs a rebase at least 😅

**Comment by bnoordhuis** [comment]
I fixed this just now in #660. Please file a new issue if anything still doesn't work as expected.

**Pull request closed without merging**